src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/src/api-authorization/authorize.interceptor.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpEvent } from '@angular/common/http';
+import { Observable, from } from 'rxjs';
+import { AuthorizeService } from './authorize.service';
+import { mergeMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthorizeInterceptor implements HttpInterceptor {
+  constructor(private authorize: AuthorizeService) { }
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return this.authorize.getAccessToken()
+      .pipe(mergeMap(token => this.processRequestWithToken(token, req, next)));
+  }
+
+  // Checks if there is an access_token available in the authorize service
+  // and adds it to the request in case it's targeted at the same origin as the
+  // single page application.
+  private processRequestWithToken(token: string, req: HttpRequest<any>, next: HttpHandler) {
+    if (!!token && this.IsSameOriginUrl(req)) {
+      req = req.clone({
+        setHeaders: {
+          Authorization: `Bearer ${token}`
+        }
+      });
+    }
+
+    return next.handle(req);
+  }
+
+  private IsSameOriginUrl(req: any) {
+    // It's an absolute url with the same origin.
+    if(req.url.startsWith(`${window.location.origin}/`)){
+      return true;
+    }
+
+    // It's a protocol relative url with the same origin.
+    // For example: //www.example.com/api/Products
+    if(req.url.startsWith(`//${window.location.host}/`)){
```

Comment:
I know auth and SSR generally don't work together seamlessly anyway, but just wanted to check: since we're assuming `window` exists here, we're accepting that this will just throw if anyone tries to add SSR support and their pages try to make `fetch` requests. Is that deliberate and accepted?

I'm not saying this is a blocker, just want to be sure it's an explicit choice.

Reply:
We use window in a couple of places. Given that we currently don't support SSR this is fine. If we wanted to enable it, we would have to look into how to do this. Probably with the location service or something similar. I don't know if you can register different services for SSR

Reply:
> I don't know if you can register different services for SSR

You can. Not saying you should change this, though.